lib/src/message/line.rs

```diff
@@ -49,31 +48,30 @@ impl<'a> From<&'a str> for Line {
 
 
 #[derive(Debug)]
-pub struct Lines<'a>(Peekable<str::Lines<'a>>);
+pub struct Lines<I, S>(Peekable<I>)
+    where I: Iterator<Item = S>, S: AsRef<str>;
 
-impl<'a> Lines<'a> {
-    pub fn new(text: &'a str) -> Lines<'a> {
-        Lines::from(text.lines())
-    }
-}
-
-impl<'a> From<str::Lines<'a>> for Lines<'a> {
-    fn from(lines: str::Lines<'a>) -> Self {
+impl<I, S> From<I> for Lines<I, S>
+    where I: Iterator<Item = S>, S: AsRef<str>
+{
+    fn from(lines: I) -> Self {
         Lines(lines.peekable())
     }
 }
 
-impl<'a> Iterator for Lines<'a> {
+impl<I, S> Iterator for Lines<I, S>
+    where I: Iterator<Item = S>, S: AsRef<str>
+{
     type Item = Line;
 
     fn next(&mut self) -> Option<Self::Item> {
-        match self.0.next().map(|line| Line::from(line)) {
+        match self.0.next().map(|line| Line::from(line.as_ref())) {
```

Comment:
Please cherry-pick https://github.com/matthiasbeyer/git-dit/commit/402121176526ea7460efba40d3ec3cef9ab4ae54 to fix this.